### PR TITLE
Update for Python API to create networks with Memory layers

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
 - *Added new ctor of ngraph::Function to python API*
 - *Updated unit tests*

### Tickets:
 - *56653*

### CODE REVIEW

File: inference-engine/ie_bridges/python/tests/test_InferRequest.py [hunk 1]
```diff
@@ -525,17 +525,16 @@ def test_resize_algorithm_work(device):
...
 
 
-# issue 56653
-@pytest.mark.skip(reason="Test will enable when nGraph Python API allows to create network with memory")
 def test_query_state(device):
     import ngraph as ng
     from ngraph.impl import Function
-    input_data = ng.parameter([5, 7], name="input_data", dtype=np.float32)
+
+    input_data = ng.parameter([10], name="input_data", dtype=np.float32)
```
Comment (nkogteva) [inline inference-engine/ie_bridges/python/tests/test_InferRequest.py:540]: Does it make sense to have tests for other shapes and precisions?

**Comment by akuporos** [comment]
I suppose you should also add a test for new function ctor. @postrational @jiwaszki could you suggest better place for it?

File: inference-engine/ie_bridges/python/tests/test_InferRequest.py [hunk 2]
```diff
@@ -544,9 +551,100 @@ def test_query_state(device):
...
+    ie_core = ie.IECore()
+    if device == "CPU":
+        if ie_core.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin")
+
+    import ngraph as ng
+    from ngraph.impl import Function
+    from openvino.inference_engine import TensorDesc, Blob
+
+    input_data = ng.parameter([10], name="input_data", dtype=np.float32)
```
Comment (akuporos) [inline inference-engine/ie_bridges/python/tests/test_InferRequest.py:620]: Let's move s network creation to another function as it was duplicated in these tests.

File: inference-engine/ie_bridges/python/tests/test_InferRequest.py (hunk 1 above)
Comment (jiwaszki) [reply inference-engine/ie_bridges/python/tests/test_InferRequest.py:540]: It makes prefect sense when we are looking at the overall test coverage and early catching of bugs. Parameterization of the test will do.

**Review by jiwaszki — CHANGES_REQUESTED** [review]
> I suppose you should also add a test for new function ctor. @postrational @jiwaszki could you suggest the best place for it?

@akuporos @itikhono The best place to test new addition of sinks is **ngraph/python/tests/test_ngraph/test_basic.py** right after `test_ngraph_function_api()`.

File: inference-engine/ie_bridges/python/tests/test_InferRequest.py (hunk 2 above)
Comment (itikhono) [reply inference-engine/ie_bridges/python/tests/test_InferRequest.py]: done

File: inference-engine/ie_bridges/python/tests/test_InferRequest.py (hunk 1 above)
Comment (itikhono) [reply inference-engine/ie_bridges/python/tests/test_InferRequest.py]: parameterized

**Review by akuporos — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**